src/components/Dropdowns/SelectionDropdown.tsx

```diff
@@ -1,9 +1,20 @@
-import { Tooltip } from "antd";
+import { SearchOutlined } from "@ant-design/icons";
+import { Input, InputRef, Tooltip } from "antd";
 import { ItemType, MenuItemType } from "antd/es/menu/hooks/useItems";
-import React, { ReactElement, useMemo } from "react";
+import Fuse from "fuse.js";
+import React, { MutableRefObject, ReactElement, useMemo, useRef, useState, useTransition } from "react";
 
+import { FlexColumn } from "../../styles/utils";
+
+import LoadingSpinner from "../LoadingSpinner";
 import AccessibleDropdown from "./AccessibleDropdown";
-import DropdownItem, { DropdownItemList } from "./DropdownItem";
+import DropdownItem from "./DropdownItem";
+import DropdownItemList from "./DropdownItemList";
+
+// TODO: Have the dropdown show a loading indicator after a selection has been made
+// but before the prop value updates. -> this is especially noticeable when slow datasets.
+// Is there a way we can do this using async promises, maybe? If the promise rejects,
+// discard the changed value?
```

Comment:
what prop value takes so long to update after a dropdown selection is made?

Reply:
[Visible in the preview link here when switching datasets](https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-244/?collection=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fmicroscopy%2FClusterOutput%2FH2B_Deliverable_AnalysisPipelineOutput%2FH2B_Deliverable_InputImages_123Timelapses_Composite_Output%2Fcolorizer%2Fcollection.json&dataset=3500005807_2&feature=R0Nuclei%20DNA%20mean%20pixel%20intensity&color=matplotlib-cool&palette-key=adobe&bg-sat=100&bg-brightness=100&fg-alpha=100&outlier-color=c0c0c0&outlier-mode=1&filter-color=dddddd&filter-mode=1&tab=track_plot&scalebar=1&timestamp=1&path=1&keep-range=0&scatter-range=all)

The datasets dropdown is the only real offender for this; I think showing a loading indicator would be helpful, but we could also try and speed up the dataset load time. It's likely taking longer because of the number of features. See #20!

Reply:
I see.  My first thought it that feels like something that should be outside of the selection dropdown's purview.  If making a selection starts loading a dataset, and loading a dataset is a long running process, the rest of the app probably needs to know about it. Anyway it's a good TODO!